staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -1971,3 +1971,54 @@ func BenchmarkCacher_GetList(b *testing.B) {
 			})
 	}
 }
+
+// TestDoNotPopExpiredWatchersWhenNoEventsSeen makes sure that
+// a bookmark event will be delivered after the cacher has seen an event.
+// Previously the watchers have been removed from the "want bookmark" queue.
+func TestDoNotPopExpiredWatchersWhenNoEventsSeen(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WatchList, true)()
+	backingStorage := &dummyStorage{}
+	cacher, _, err := newTestCacher(backingStorage)
+	if err != nil {
+		t.Fatalf("Couldn't create cacher: %v", err)
+	}
+	defer cacher.Stop()
+
+	// wait until cacher is initialized.
+	if err := cacher.ready.wait(context.Background()); err != nil {
+		t.Fatalf("unexpected error waiting for the cache to be ready")
+	}
+
+	opts := storage.ListOptions{Predicate: storage.Everything}
+	opts.Predicate.AllowWatchBookmarks = true
+	opts.SendInitialEvents = pointer.Bool(true)
+	w, err := cacher.Watch(context.Background(), "pods/ns", opts)
+	require.NoError(t, err, "failed to create watch: %v")
+	defer w.Stop()
+
+	// the default bookmarkTimer is ~1 sec, waiting 2 seconds is enough
+	time.Sleep(2 * time.Second)
+
+	makePod := func(rv uint64) *example.Pod {
```

Comment:
Hmm - I'm wondering why do we need this.

1. watchcache is initialized from the dummystorage list with return RV=100
2. then we initialize watch, it goes to storage and returns also RV=100
3. so given it is already synchronized, it should return immediately

What am I missing?

Reply:
heh... I'm stupid.

I suggest the following:
1) update the comment above (see my other comment)
2) Here I would ideally switch to imitate progress-notify, but unfortunately this isn't easy...
   So let's leave as is just add a comment like
```
// Send an even to ensure that lastProcessedResourceVersion in Cacher will change to non-zero value.
```

Reply:
:)